OpenRA.Mods.Common/Traits/ExplosionOnDamageTransition.cs

```diff
@@ -45,12 +47,13 @@ public void RulesetLoaded(Ruleset rules, ActorInfo ai)
 		}
 	}
 
-	public class ExplosionOnDamageTransition : INotifyDamageStateChanged
+	public class ExplosionOnDamageTransition : ConditionalTrait<ExplosionOnDamageTransitionInfo>, INotifyDamageStateChanged
 	{
 		readonly ExplosionOnDamageTransitionInfo info;
```

Comment:
This can be removed, and the lines below use `Info` from the base class instead.